core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -48,20 +123,85 @@ import Pull._
   *
   * `raiseError` is caught by `handleErrorWith`:
   *   - `handleErrorWith(raiseError(e))(f) == f(e)`
+  *
+  * @tparam F[_] the type of functional effects that can be performed by this pull.
+  * @tparam O The type of outputs emitted by this Pull.
+  * @tparam R The type of carry-over result that is returned by this Pull, if it returns successfully.
   */
 sealed abstract class Pull[+F[_], +O, +R] {
 
-  /** Alias for `_.map(_ => o2)`. */
-  def as[R2](r2: R2): Pull[F, O, R2] = map(_ => r2)
+  /** Applies the result of this pull to `f` and returns the result.
+    *
+    * This method returns a new composed pull. The semantics of this pull
+    *
+    * - If `this` pull completes successfully with a result of type R, 
+    *   the `f` function is applied to the result to build a new Pull, 
+    *   and compilation continues by running that pull. The end return of 
```

Comment:
Similar to earlier comment -- no need to mention compilation when describing the API elements.